src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -55,17 +59,45 @@ export const sortTokens = ({
  * In case of a tie, it sorts by token name and network name.
  *
  * @param $tokens - The list of tokens to sort.
+ * @param $balancesStore - The balances data for the tokens.
+ * @param $exchanges - The exchange rates data for the tokens.
  * @returns The sorted list of tokens.
  *
  */
-export const pinTokensWithBalanceAtTop = ($tokens: TokenUi[]): TokenUi[] => {
+export const pinTokensWithBalanceAtTop = ({
+	$tokens,
+	$balances,
+	$exchanges
+}: {
+	$tokens: Token[];
+	$balances: CertifiedStoreData<BalancesData>;
+	$exchanges: ExchangesData;
+}): TokenUi[] => {
 	const [positiveBalances, nonPositiveBalances] = $tokens.reduce<[TokenUi[], TokenUi[]]>(
-		(acc, token) => (
-			(token.usdBalance ?? 0) > 0 || (token.balance ?? ZERO).gt(0)
-				? acc[0].push(token)
-				: acc[1].push(token),
-			acc
-		),
+		(acc, token) => {
+			const balance: BigNumber | undefined = $balances?.[token.id]?.data;
+			const exchangeRate: number | undefined = $exchanges?.[token.id]?.usd;
+
+			const usdBalance: number | undefined = nonNullish(exchangeRate)
+				? usdValue({
+						token,
+						balance,
+						exchangeRate
+					})
+				: undefined;
+
+			const tokenUI: TokenUi = {
+				...token,
+				balance,
+				usdBalance
+			};
+
+			(usdBalance ?? 0) > 0 || (balance ?? ZERO).gt(0)
```

Comment:
Not necessarely for this PR and a bit an feedback but, given that the function isn't a single line anymore, do we have chance here to introduce a `if` and transform the `push` into a more functional programming approach?

Reply:
i will do it in this one

Reply:
@peterpeterparker before I merge, have a look, if that what you meant

Reply:
Almost there, still not fully functional. We can keep the push in this PR and try to do that later or not.